### ISSUE TITLE
dorefa-net convergence in a loacal optimal, training error keep in a high level ?

### ISSUE THREAD

**Issue by silverlining21** [ppwwyyxx/tensorpack#73]
while I'm training dorefa-net on my own data set. I got training error keep in a very high level which top one `training error = 16%` for a long time since `iterations=270k` ? here is my result and experiment setup:

**result** 
![qq 20161223101815](https://cloud.githubusercontent.com/assets/8264748/21445863/d312ad26-c8f9-11e6-8e93-d045507040f1.png)

**setup** 
1. **With (W,A,G)=(32,32,32)** 

1. learning rate
 ```
    starter_learning_rate = 1e-2
    end_learning_rate = 1e-5
    total_steps = 3e5
    lr = tf.train.polynomial_decay(
        starter_learning_rate,
        global_step=get_global_step_var(),
        decay_steps=total_steps,
        end_learning_rate=end_learning_rate,
        power=0.5)
```
3. input 
![image](https://cloud.githubusercontent.com/assets/8264748/21445883/077f7904-c8fa-11e6-82f7-5e4f4a85652b.png)

4. network 
![image](https://cloud.githubusercontent.com/assets/8264748/21445887/1b9067c8-c8fa-11e6-86bd-a19f1af622d1.png)

5. Another experiment **With (W,A,G)=(1,2,32),** other setup keep the same with above. I got the result like that: after nearly `600k interactions`, it's show NO signal to convergence.

    ![image](https://cloud.githubusercontent.com/assets/8264748/21445960/6b45f4e4-c8fb-11e6-88d8-cb61be6db36f.png)

Do you have any ideas about the reasons of that? thanks in advance.

..

**Comment by silverlining21** [comment]
I run these two experiment at the same time on different GUPs, does this effect anything ?

**Comment by silverlining21** [comment]
the first experiment, which  **(W,A,G)=(32,32,32)**, after I change the learning rate to 0.01 and restore training and I got a much better convergence at `top_1_error=0.06`. but for the  **(W,A,G)=(1,2,32)** it's still the same. :sob:

**Comment by ppwwyyxx** [comment]
Issues are for potential problems with this library. Since you're training with your code, there are a lot of things that could go wrong and we don't have time to look through it.

If you have 6k classes why do you think the error rate you got is not good? If you have a better model of some other architectures, why don't you make changes starting from that model? 

From the information you provided, there are also issues in your architecture that are not consistent with how DoReFa-Net is designed. You quantized the activations before bn0 which doesn't have much run-time benefits but hurt performance. Activations before fct is also quantized but we prefer not to do this as discussed in the paper.

**Comment by silverlining21** [comment]
thanks!
